src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -35,6 +40,224 @@ public ProjectView()
+                return _packageDetailControlModel;
+            }
+            set
+            {
+                _packageDetailControlModel = value;
+            }
+        }
+
+        private DetailControlModel _detailModel;
+
+        private DetailControlModel DetailModel
+        {
+            get
+            {
+                if (_detailModel == null) _detailModel = (DetailControlModel)DataContext;
+                return _detailModel;
+            }
+            set
+            {
+                _detailModel = value;
+            }
+        }
+
+        private string _previousFilter;
+        private string PreviousText
+        {
+            get
+            {
+                if (_previousFilter != null) return _previousFilter;
+                return string.Empty;
+            }
+            set
+            {
+                _previousFilter = value;
+            }
+        }
+
+        protected override void OnPreviewKeyDown(KeyEventArgs e)
+        {
+            if (PackageDetailControlModel.IsProjectPackageReference)
+            {
+                switch (e.Key)
+                {
+                    case Key.Tab:
+                    case Key.Enter:
+                        _versions.IsDropDownOpen = false;
+                        break;
+                    case Key.Escape:
+                        _versions.IsDropDownOpen = false;
+                        _versions.SelectedIndex = -1;
+                        break;
+                    case Key.Down:
+                        if (_versions.SelectedIndex < _versions.Items.Count - 1)
+                        {
+                            // Handle null separator
+                            if (_versions.Items[_versions.SelectedIndex + 1] == null)
+                            {
+                                _versions.SelectedIndex = _versions.SelectedIndex + 2;
+                            }
+                            else
+                            {
+                                _versions.SelectedIndex++;
+                            }
+                            PreviousText = _versions.Text;
+
+                            e.Handled = true;
+                        }
+                        break;
+                    case Key.Up:
+                        if (_versions.SelectedIndex > 0)
+                        {
+                            // Handle null separator
+                            if (_versions.Items[_versions.SelectedIndex - 1] == null)
+                            {
+                                _versions.SelectedIndex = _versions.SelectedIndex - 2;
+                            }
+                            else
+                            {
+                                _versions.SelectedIndex--;
+                            }
+                            PreviousText = _versions.Text;
+
+                            e.Handled = true;
+                        }
+                        break;
+                    default:
```

Comment:
Maybe we ignore spaces entirely like:
```
case key.space
{ e.handled = true }
```

?

What if I copy/paste text that has a space, will it trim out the space, or leave it, or explode 💣?
paste "( 1.0,  2.0 )" => combobox text becomes "(1.0.0,2.*)"